packages/@lwc/template-compiler/src/shared/types.ts

```diff
@@ -4,15 +4,20 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-import * as babelTypes from '@babel/types';
 import * as parse5 from 'parse5-with-errors';
 import { CompilerDiagnostic } from '@lwc/errors';
 
-export type TemplateIdentifier = babelTypes.Identifier;
+export type TemplateIdentifier = { type: 'Identifier'; name: string };
```

Comment:
We aren't using `estree` types here because this is where we define our own types?

```
export interface Identifier extends BaseNode, BaseExpression, BasePattern {
  type: "Identifier";
  name: string;
}
```

Reply:
Because of the restrictive nature of the template compiler, the template compiler only accepts a subset of `MemberExpression`. I decided to expose only a subset of the actual properties to make sure that the types are in line with the parser restrictions. This avoids creating unnecessary type guards in the code gen.